lib/tcp.js

```diff
@@ -157,7 +157,10 @@ exports.client = function (options, transportUtil) {
       reconnect.on('disconnect', function (err) {
         seneca.log.debug('client', type, 'disconnect', spec, topic, clientOptions,
            (err && err.stack) || err)
-        _.find(connections, { clientOptions: clientOptions }).setup = false
+        var connections = _.find(connections, { clientOptions: clientOptions });
```

Comment:
The seneca-service bootstrap looked like this.
```
// service-x.js
const seneca = require('seneca')(
    {
        log: {
            level: 'info+'
        }
    }
)

/* extensions */
seneca.use('basic').use('entity')
seneca.use('mem-store')

/* start, listen */
seneca.listen(
    {
        type: 'tcp',
        port: process.env.PORT || 39999,
        host: process.env.HOST || '0.0.0.0',
        timeout: 2500
    }
)
```

The client looked like this. We have used seneca with restify. But not as a restify middleware. 
```
// contact-service-x.js
const service = seneca.client(
    {
        type: 'tcp',
        port: process.env.SERVICE_X_PORT,
        host: process.env.SERVICE_X_HOST
    }
)

module.exports = service
```
Restify route handler
```
// route-x.js
const service = require('./contact-service-x.js');

service.act(
    {
        role: 'service-x',
        cmd: 'action',
        data: []
    },
    handleServiceResponse
)
```